### PR TITLE
feat: combined model + script artifact

### PR DESCRIPTION
**Pull request by evakravi**
*Issue #, if available:*

*Description of changes:*
`model_uris.retrieve` returns a combined model + script artifact if it's found in metadata for model. 

*Testing done:*

## Merge Checklist

_Put an `x` in the boxes that apply. You can also fill these out after creating the PR. If you're unsure about any of them, don't hesitate to ask. We're here to help! This is simply a reminder of what we are going to look for before merging your pull request._

#### General

- [ ] I have read the [CONTRIBUTING](https://github.com/aws/sagemaker-python-sdk/blob/master/CONTRIBUTING.md) doc
- [ ] I certify that the changes I am introducing will be backward compatible, and I have discussed concerns about this, if any, with the Python SDK team
- [ ] I used the commit message format described in [CONTRIBUTING](https://github.com/aws/sagemaker-python-sdk/blob/master/CONTRIBUTING.md#committing-your-change)
- [ ] I have passed the region in to all S3 and STS clients that I've initialized as part of this change.
- [ ] I have updated any necessary documentation, including [READMEs](https://github.com/aws/sagemaker-python-sdk/blob/master/README.rst) and [API docs](https://github.com/aws/sagemaker-python-sdk/tree/master/doc) (if appropriate)

#### Tests

- [ ] I have added tests that prove my fix is effective or that my feature works (if appropriate)
- [ ] I have added unit and/or integration tests as appropriate to ensure backward compatibility of the changes
- [ ] I have checked that my tests are not configured for a specific region or account (if appropriate)
- [ ] I have used [`unique_name_from_base`](https://github.com/aws/sagemaker-python-sdk/blob/master/src/sagemaker/utils.py#L77) to create resource names in integ tests (if appropriate)

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.

### CODE REVIEW

File: src/sagemaker/jumpstart/artifacts.py [hunk 1]
```diff
@@ -205,6 +208,8 @@ def _retrieve_model_uri(
         VulnerableJumpStartModelError: If any of the dependencies required by the script have
             known security vulnerabilities.
         DeprecatedJumpStartModelError: If the version of the model is deprecated.
+        NotImplementedError: If the combination of arguments doesn't support combined model
```
Comment (whittech1) [inline src/sagemaker/jumpstart/artifacts.py:211]: Could we use `ValueError` to indicate that the provided arg values are invalid?

File: src/sagemaker/jumpstart/types.py [hunk 2]
```diff
@@ -293,6 +293,7 @@ class JumpStartModelSpecs(JumpStartDataHolderType):
         "training_vulnerabilities",
         "deprecated",
         "metrics",
+        "hosting_prepacked_artifact_key",
```
Comment (whittech1) [inline src/sagemaker/jumpstart/types.py:296]: Would you be opposed to adding the `training_prepacked_script_key` in this PR along with similar logic for retrieval?

Comment (evakravi) [reply src/sagemaker/jumpstart/types.py:296]: Let's do that in another PR

File: src/sagemaker/jumpstart/artifacts.py [hunk 3]
```diff
@@ -218,10 +221,24 @@ def _retrieve_model_uri(
         tolerate_deprecated_model=tolerate_deprecated_model,
     )
 
+    error_msg_no_combined_artifact = (
+        "No combined script + model tarball available "
```
Comment (JGuinegagne) [inline src/sagemaker/jumpstart/artifacts.py:225]: nit: `No combined script and model tarball [...]`

**Review by JGuinegagne — APPROVED** [review]

**Review by whittech1 — APPROVED** [review]

**Review by claytonparnell — COMMENTED** [review]
/bot run unit-tests

**Pull request merged**